pkg/segment/results/mresults/seriesresult.go

```diff
@@ -790,6 +795,62 @@ func evaluate(ts map[uint32]float64, mathFunc float64Func) {
 	}
 }
 
+type uint32Func func(uint32) float64
+
+func evaluateTimeFunc(allDPs map[uint32]float64, timeFunc uint32Func) {
+	for dpTs := range allDPs {
+		allDPs[dpTs] = timeFunc(dpTs)
+	}
+}
+
+func ApplyTimeFunction(allDPs map[uint32]float64, function structs.Function) (map[uint32]float64, error) {
+	switch function.TimeFunction {
+	case segutils.Hour:
+		evaluateTimeFunc(allDPs, func(dpTs uint32) float64 {
+			t := time.Unix(int64(dpTs), 0).UTC()
+			return float64(t.Hour())
+		})
+	case segutils.Minute:
+		evaluateTimeFunc(allDPs, func(dpTs uint32) float64 {
+			t := time.Unix(int64(dpTs), 0).UTC()
+			return float64(t.Minute())
+		})
```

Comment:
nit: take this one step further to reduce duplicate code; do `t := time.Unix(int64(dpTs), 0).UTC()` inside `evaluateTimeFunc()` instead of having that line every time you call `evaluateTimeFunc()`